auth/src/main/java/com/homihq/db2rest/auth/basic/BasicAuthProvider.java

```diff
@@ -41,15 +42,14 @@ public UserDetail authenticate(HttpServletRequest request) {
         String username = parts[0];
         String password = parts[1];
 
-
-        Optional<User> user = authDataProvider.getUserByUsername(username);
-
-        if(user.isPresent() && StringUtils.equals(password, user.get().password())) {
-            return new UserDetail(username, user.get().roles());
+        if(authDataProvider instanceof FileAuthDataProvider fileAuthDataProvider) {
```

Comment:
We should depend on the **authProvider** abstraction with a detailed implementation defined in the **AuthConfiguration** like a factory. Let's imagine we are going to add one more provider. After that, you should update your code in several places. Let's stay as is and depend on the **AuthDataProvider** abstraction.